protostar/commands/migrate/migrate_command.py

```diff
@@ -0,0 +1,146 @@
+                type="path",
+            ),
+            Command.Argument(
+                name="rollback",
+                description="Run `rollback` function in the migration script.",
+                type="bool",
+            ),
+            Command.Argument(
+                name="no-confirm",
+                description="Skip confirming building the project.",
+                type="bool",
+            ),
+            Command.Argument(
+                name=MigrateCommand.GATEWAY_URL_ARG_NAME,
+                description="The URL of a StarkNet gateway. It is required unless `--network` is provided.",
+                type="str",
+            ),
+            Command.Argument(
+                name=MigrateCommand.NETWORK_ARG_NAME,
+                short_name="n",
+                description=(
+                    "\n".join(
+                        [
+                            "The name of the StarkNet network.",
+                            "It is required unless `--gateway-url` is provided.",
+                            "",
+                            "Supported StarkNet networks:",
+                        ]
+                        + [f"- `{n}`" for n in NetworkConfig.get_starknet_networks()]
+                    )
+                ),
+                type="str",
+            ),
+        ]
+
+    async def run(self, args):
+        await self.migrate(
+            migration_file_path=args.path,
+            rollback=args.rollback,
+            gateway_url=args.gateway_url,
+            network=args.network,
+            output_dir_path=args.output_dir,
+            no_confirm=args.no_confirm,
+        )
+
+    # pylint: disable=too-many-arguments
+    async def migrate(
+        self,
+        migration_file_path: Path,
+        rollback: bool,
+        gateway_url: Optional[str],
+        network: Optional[str],
+        output_dir_path: Optional[Path],
+        no_confirm: bool,
+    ):
+        if network is None and gateway_url is None:
+            raise ProtostarException(
+                f"Argument `{MigrateCommand.GATEWAY_URL_ARG_NAME}` or `{MigrateCommand.NETWORK_ARG_NAME}` is required"
+            )
+
+        network_config = NetworkConfig.build(gateway_url, network)
+
+        # mitigates the risk of running migrate on an outdated project
+        should_confirm = not no_confirm
+        if should_confirm and not self._requester.confirm(
+            "Did you build the project before running this command?"
+        ):
+            self._logger.info("Migration cancelled")
+            return
+
+        self._migrator_factory.set_logger(self._logger, self._log_color_provider)
+
+        migrator = await self._migrator_factory.build(
+            migration_file_path,
+            config=Migrator.Config(gateway_url=network_config.gateway_url),
+        )
+
+        try:
+            migrator_history = await migrator.run(
+                mode="down" if rollback else "up",
```

Comment:
could this be an enum instead of literal strings?

Reply:
Would be consistent